### PR TITLE
Feature window android

### PR DESCRIPTION
**Pull request by Sumenia**
Deleting android/log until we have the cross platform logger
Deleting unnecessary events until we have uniform cross platform events

### CODE REVIEW

File: samples/window/CMakeLists.txt [hunk 1]
```diff
@@ -1,22 +1,48 @@
...
+
+if(LUG_COMPILER_MSVC)
+    if(CMAKE_CXX_FLAGS MATCHES "/W[0-4]")
+        string(REGEX REPLACE "/W[0-4]" "/W4" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+    else()
+        list(APPEND CMAKE_CXX_FLAGS "/W4")
+    endif()
+elseif(LUG_COMPILER_GCC OR LUG_COMPILER_CLANG)
+    list(APPEND CMAKE_CXX_FLAGS "-Wall -Wextra -Werror -std=c++14")
+    string(REPLACE ";" " " "CMAKE_CXX_FLAGS" "${CMAKE_CXX_FLAGS}")
```
Comment (Quentin01) [inline samples/window/CMakeLists.txt]: Do that outside of the if. See https://github.com/Lugdunum3D/Lugdunum/blob/dev/samples/base/CMakeLists.txt#L30

File: samples/window/CMakeLists.txt [hunk 2]
```diff
@@ -1,22 +1,48 @@
...
-if(LUG_FOUND)
-    include_directories(${LUG_INCLUDE_DIR})
+
+include_directories(${LUG_INCLUDE_DIR})
+
+if(LUG_OS_ANDROID)
+    add_library(${EXECUTABLE_NAME}
+            SHARED
+            main.cpp
+            )
```
Comment (Quentin01) [inline samples/window/CMakeLists.txt]: Bad indentation

File: samples/window/CMakeLists.txt [hunk 3]
```diff
@@ -1,22 +1,48 @@
...
+
+    target_link_libraries(${EXECUTABLE_NAME}
+            log
+            android
+            ${LUG_LIBRARIES}
+            )
```
Comment (Quentin01) [inline samples/window/CMakeLists.txt]: Bad indentation

File: samples/window/main.cpp [hunk 4]
```diff
@@ -1,5 +1,5 @@
-#include <lug/Config.hpp>
 #include <lug/Window/Window.hpp>
+#include <string>
```
Comment (Quentin01) [inline samples/window/main.cpp]: Remove that include

File: samples/window/main.cpp [hunk 5]
```diff
@@ -15,11 +15,9 @@ int main() {
             if (event.type == lug::Window::EventType::CLOSE) {
                 window->close();
             }
-
```
Comment (Quentin01) [inline samples/window/main.cpp:18]: Why have you remove that line?

File: samples/window/main.cpp [hunk 6]
```diff
@@ -15,11 +15,9 @@ int main() {
...
         }
 
         // ...
     }
-
```
Comment (Quentin01) [inline samples/window/main.cpp:23]: Why have you remove that line?

File: src/lug/Main/MainAndroid.hpp [hunk 7]
```diff
@@ -2,15 +2,19 @@
...
 #include <android/native_activity.h>
+#include <lug/Window/Window.hpp>
 
 extern int main(int argc, const char* argv[]);
 
 namespace lug {
 namespace Main {
 
-class LugAndroidApp {
+extern std::queue<lug::Window::Event> events;
```
Comment (Quentin01) [inline src/lug/Main/MainAndroid.hpp]: Why are you declaring them extern here? `MainAndroid.cpp` doesn't need that.

File: src/lug/Window/Android/WindowImplAndroid.cpp [hunk 8]
```diff
@@ -0,0 +1,39 @@
+#include <lug/Window/Android/WindowImplAndroid.hpp>
+
+namespace lug {
+namespace Window {
+namespace priv {
+
+lug::Window::priv::WindowImpl::WindowImpl(Window*) {}
```
Comment (Quentin01) [inline src/lug/Window/Android/WindowImplAndroid.cpp]: You don't need to specify `lug::Window::priv::` here

File: src/lug/Window/Android/WindowImplAndroid.cpp [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+bool lug::Window::priv::WindowImpl::create(const std::string&, Style) {
```
Comment (Quentin01) [inline src/lug/Window/Android/WindowImplAndroid.cpp]: You don't need to specify `lug::Window::priv::` here

File: src/lug/Window/Android/WindowImplAndroid.cpp [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    return true;
+}
+
+void lug::Window::priv::WindowImpl::close() {}
```
Comment (Quentin01) [inline src/lug/Window/Android/WindowImplAndroid.cpp]: You don't need to specify `lug::Window::priv::` here

File: src/lug/Window/Android/WindowImplAndroid.cpp [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+
+bool lug::Window::priv::WindowImpl::pollEvent(lug::Window::Event& event) {
```
Comment (Quentin01) [inline src/lug/Window/Android/WindowImplAndroid.cpp]: You don't need to specify `lug::Window::priv::` here

File: src/lug/Window/Android/WindowImplAndroid.cpp [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+
+    if (lug::Main::inputQueue != nullptr) {
+        AInputEvent *androidEvent = nullptr;
+        while (AInputQueue_getEvent(lug::Main::inputQueue, &androidEvent) >= 0) {
+            if (AInputQueue_preDispatchEvent(lug::Main::inputQueue, androidEvent)) {
+                continue;
+            }
+            int32_t handled = 0;
+            AInputQueue_finishEvent(lug::Main::inputQueue, androidEvent, handled);
```
Comment (Quentin01) [inline src/lug/Window/Android/WindowImplAndroid.cpp]: Why not `AInputQueue_finishEvent(lug::Main::inputQueue, androidEvent, 0);` ?

File: src/lug/Window/Android/WindowImplAndroid.cpp [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+
+bool lug::Window::priv::WindowImpl::pollEvent(lug::Window::Event& event) {
+
+    if (lug::Main::inputQueue != nullptr) {
+        AInputEvent *androidEvent = nullptr;
+        while (AInputQueue_getEvent(lug::Main::inputQueue, &androidEvent) >= 0) {
+            if (AInputQueue_preDispatchEvent(lug::Main::inputQueue, androidEvent)) {
+                continue;
+            }
+            int32_t handled = 0;
```
Comment (Quentin01) [inline src/lug/Window/Android/WindowImplAndroid.cpp]: Maybe one vertical space?

File: src/lug/Window/Android/WindowImplAndroid.cpp [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+    return true;
+}
+
+void lug::Window::priv::WindowImpl::close() {}
+
+bool lug::Window::priv::WindowImpl::pollEvent(lug::Window::Event& event) {
+
+    if (lug::Main::inputQueue != nullptr) {
+        AInputEvent *androidEvent = nullptr;
+        while (AInputQueue_getEvent(lug::Main::inputQueue, &androidEvent) >= 0) {
```
Comment (Quentin01) [inline src/lug/Window/Android/WindowImplAndroid.cpp]: Maybe one vertical space?

File: src/lug/Window/Android/WindowImplAndroid.hpp [hunk 15]
```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <lug/Window/Window.hpp>
+#include <lug/Main/MainAndroid.hpp>
```
Comment (Quentin01) [inline src/lug/Window/Android/WindowImplAndroid.hpp]: Do you really need that include? Can't you declare your extern here?

File: src/lug/Window/Android/WindowImplAndroid.hpp [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+
+class WindowImpl {
+public:
+    WindowImpl(Window*);
+    ~WindowImpl() = default;
+
+    bool create(const std::string& title, Style style);
+    void close();
+    bool pollEvent(lug::Window::Event& event);
+private:
```
Comment (Quentin01) [inline src/lug/Window/Android/WindowImplAndroid.hpp]: Remove that `private` if it isn't used

File: src/lug/Window/Android/WindowImplAndroid.hpp [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+};
+
+} // namespace priv
+} // namespace Window
+} // namespace lug
```
Comment (Quentin01) [inline src/lug/Window/Android/WindowImplAndroid.hpp]: New line at end of file

File: src/lug/Window/CMakeLists.txt [hunk 18]
```diff
@@ -16,16 +16,22 @@ set(INC
...
 
-    find_package(X11 REQUIRED)
-    set(LUG_WINDOW_LIBRARIES ${X11_LIBRARIES})
-
 elseif(LUG_OS_WINDOWS)
     list(APPEND INC ${SRCROOT}/Win32/WindowImplWin32.hpp)
     list(APPEND SRC ${SRCROOT}/Win32/WindowImplWin32.cpp)
+elseif(LUG_OS_ANDROID)
+    include_directories(${CMAKE_SYSROOT}/usr/include)
+    list(APPEND EXT_LIBRARIES android)
```
Comment (Quentin01) [inline src/lug/Window/CMakeLists.txt]: Can't you append the two items in one line?

**Review by Quentin01 — CHANGES_REQUESTED** [review]

File: src/lug/Main/MainAndroid.hpp (hunk 7 above)
Comment (Sumenia) [reply src/lug/Main/MainAndroid.hpp]: I spent 1 hour trying to compile on android with the extern in lug/Window/Android/WindowImplAndroid.cpp with no success , ended up putting in here and it works

File: samples/window/CMakeLists.txt (hunk 1 above)
Comment (Sumenia) [reply samples/window/CMakeLists.txt]: I took the code from base at that time, we edited it after but indeed I didn't apply the change here

File: samples/window/main.cpp (hunk 5 above)
Comment (Sumenia) [reply samples/window/main.cpp:18]: what is the need of using a vertical space here ?

Comment (Quentin01) [reply samples/window/main.cpp:18]: I don't know, but okay, this one is really useless

**Review by Quentin01 — APPROVED** [review]

**Pull request merged**